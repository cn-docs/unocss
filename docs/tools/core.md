---
title: 核心
description: UnoCSS 的核心引擎，不包含任何预设。它可以作为你自己原子 CSS 框架的引擎使用。
---

# 核心

UnoCSS 的核心引擎（无预设）：`@unocss/core`。它可以作为你自己原子 CSS 框架的引擎使用。

## 安装

::: code-group

```bash
pnpm add -D @unocss/core
```

```bash
yarn add -D @unocss/core
```

```bash
npm install -D @unocss/core
```

:::

## 使用

```ts
import { createGenerator } from '@unocss/core'

const generator = await createGenerator(
  { /* 用户选项 */ },
  { /* 默认选项 */ }
)

const { css } = await generator.generate(code)
```

## 许可证

- MIT 许可证 &copy; 2021-PRESENT [Anthony Fu](https://github.com/antfu)
