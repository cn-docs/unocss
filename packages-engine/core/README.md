# @unocss/core

The core engine of [UnoCSS](https://github.com/unocss/unocss) without any presets. It can be used as the engine of your own atomic CSS framework.

## Documentation

Please refer to the [documentation](https://unocss.dev/tools/core).

## License

MIT License &copy; 2021-PRESENT [Anthony Fu](https://github.com/antfu)
