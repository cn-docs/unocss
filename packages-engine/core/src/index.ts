export * from './config'
export * from './constants'
export * from './extractors'
export * from './generator'
export * from './types'
export * from './utils'
