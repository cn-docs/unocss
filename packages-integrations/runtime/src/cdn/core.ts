import init from '../index'

init()
