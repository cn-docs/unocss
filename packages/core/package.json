{
  "name": "@unocss/core",
  "type": "module",
  "version": "0.62.3",
  "description": "The instant on-demand Atomic CSS engine.",
  "author": "Anthony Fu <anthonyfu117@hotmail.com>",
  "license": "MIT",
  "funding": "https://github.com/sponsors/antfu",
  "homepage": "https://github.com/unocss/unocss/tree/main/packages/core#readme",
  "repository": {
    "type": "git",
    "url": "https://github.com/unocss/unocss",
    "directory": "packages/core"
  },
  "bugs": {
    "url": "https://github.com/unocss/unocss/issues"
  },
  "keywords": [
    "unocss",
    "atomic-css",
    "atomic-css-engine",
    "css",
    "tailwind",
    "windicss"
  ],
  "sideEffects": false,
  "exports": {
    ".": {
      "types": "./dist/index.d.mts",
      "default": "./dist/index.mjs"
    }
  },
  "main": "dist/index.mjs",
  "module": "dist/index.mjs",
  "types": "dist/index.d.ts",
  "files": [
    "dist"
  ],
  "scripts": {
    "build": "unbuild",
    "stub": "unbuild --stub"
  },
  "devDependencies": {
    "magic-string": "catalog:",
    "unconfig": "catalog:"
  }
}
