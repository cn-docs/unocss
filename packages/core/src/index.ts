export * from './config'
export * from './extractors'
export * from './generator'
export * from './types'
export * from './utils'
