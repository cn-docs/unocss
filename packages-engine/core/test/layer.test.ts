import { createGenerator } from '@unocss/core'
import presetMini from '@unocss/preset-mini'
import presetWebFonts from '@unocss/preset-web-fonts'
import { describe, expect, it } from 'vitest'

describe('layers', () => {
  it('static', async () => {
    const uno = await createGenerator({
      rules: [
        ['a', { name: 'bar1' }, { layer: 'a' }],
        ['b', { name: 'bar2' }, { layer: 'b' }],
        [/^c(\d+)$/, ([, d]) => ({ name: d }), { layer: 'c' }],
        [/^d(\d+)$/, ([, d]) => `/* RAW ${d} */`, { layer: 'd' }],
      ],
      shortcuts: [
        ['abcd', 'a b c2 d3', { layer: 's' }],
        ['abc', 'a b c4'],
      ],
      presets: [],
    })
    const { css } = await uno.generate('a b abc abcd d4 c5', { preflights: false })
    expect(css).toMatchSnapshot()
  })

  it('@import layer', async () => {
    const uno = await createGenerator({
      presets: [
        presetMini(),
        presetWebFonts({
          provider: 'google',
          fonts: {
            mono: ['Fira Code', 'Fira Mono:400,700'],
          },
          inlineImports: false,
        }),
      ],
    })
    const { css } = await uno.generate('font-mono', { preflights: true })
    expect(css).toMatchInlineSnapshot(`
      "/* layer: imports */
      @import url('https://fonts.googleapis.com/css2?family=Fira+Code&family=Fira+Mono:wght@400;700&display=swap');
      /* layer: preflights */
      *,::before,::after{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgb(0 0 0 / 0);--un-ring-shadow:0 0 rgb(0 0 0 / 0);--un-shadow-inset: ;--un-shadow:0 0 rgb(0 0 0 / 0);--un-ring-inset: ;--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgb(147 197 253 / 0.5);}::backdrop{--un-rotate:0;--un-rotate-x:0;--un-rotate-y:0;--un-rotate-z:0;--un-scale-x:1;--un-scale-y:1;--un-scale-z:1;--un-skew-x:0;--un-skew-y:0;--un-translate-x:0;--un-translate-y:0;--un-translate-z:0;--un-ring-offset-shadow:0 0 rgb(0 0 0 / 0);--un-ring-shadow:0 0 rgb(0 0 0 / 0);--un-shadow-inset: ;--un-shadow:0 0 rgb(0 0 0 / 0);--un-ring-inset: ;--un-ring-offset-width:0px;--un-ring-offset-color:#fff;--un-ring-width:0px;--un-ring-color:rgb(147 197 253 / 0.5);}
      /* layer: default */
      .font-mono{font-family:"Fira Code","Fira Mono",ui-monospace,SFMono-Regular,Menlo,Monaco,Consolas,"Liberation Mono","Courier New",monospace;}"
    `)
  })
})
