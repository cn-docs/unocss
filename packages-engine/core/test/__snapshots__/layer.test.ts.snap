// Vitest Snapshot v1, https://vitest.dev/guide/snapshot.html

exports[`layers > static 1`] = `
"/* layer: shortcuts */
.abc{name:bar1;name:bar2;name:4;}
/* layer: a */
.a{name:bar1;}
/* layer: b */
.b{name:bar2;}
/* layer: c */
.c5{name:5;}
/* layer: d */
/* RAW 4 */
/* RAW 3 */
/* layer: s */
.abcd{name:bar1;name:bar2;name:2;}"
`;
